packages/react-component-event-listener/src/EventListener.ts

```diff
@@ -1,16 +1,21 @@
 import * as React from 'react'
 
 import addEventListener from './lib/addEventListener'
-import { EventListenerProps, listenerPropTypes } from './types'
 import removeEventListener from './lib/removeEventListener'
+import shouldUpdateListener from './lib/shouldUpdateListener'
+import { EventListenerProps, listenerPropTypes } from './types'
 
-class EventListener extends React.PureComponent<EventListenerProps> {
+class EventListener extends React.Component<EventListenerProps> {
   static displayName = 'EventListener'
   static propTypes = listenerPropTypes
   static defaultProps = {
     capture: false,
   }
 
+  shouldComponentUpdate(nextProps: EventListenerProps) {
+    return shouldUpdateListener(this.props, nextProps)
```

Comment:
This sounds strange, but will work because React will call actual `this.props.listener`, added UTs confirm that. This behavior is covered there: https://overreacted.io/a-complete-guide-to-useeffect/#each-render-has-its-own-everything

It will be more clear when we will migrate this logic to hooks.

Reply:
might refer to another section from the same blog that warns against using `shouldUpdate` methods https://overreacted.io/writing-resilient-components/#dont-stop-the-data-flow-in-optimizations :) For example, now, with these changes being made, we should refrain from passing `listener` value any further down (the fact that we are not rendering anything helps, but there are no explicit constrains on this component in these regards) - as it will be another source of problems.

At the same time, agree that Hooks policy around _when_ subscription happens (and, surely, around specifying update dependencies) should help in this case.

Reply:
Fully agree with you there, however should mention that `EventListener` doesn't render any children, so actually we don't stop there nothing.

And yep, hooks will be more obvious with their _when_ 👍